src/orchestrator/orchestrator.rs

```diff
@@ -0,0 +1,191 @@
+
+pub struct Orchestrator<const N_ASSETS: usize, const N_BYTES: usize> {
+    executor_spawner: Box<dyn ExecutorSpawner>,
+    entry_csvs: Vec<String>,
+}
+
+impl<const N_ASSETS: usize, const N_BYTES: usize> Orchestrator<N_ASSETS, N_BYTES> {
+    fn new(executor_spawner: Box<dyn ExecutorSpawner>, entry_csvs: Vec<String>) -> Self {
+        Self {
+            executor_spawner,
+            entry_csvs,
+        }
+    }
+
+    // Calculate the range of tasks to be assigned to a executor.
+    //
+    // * `executor_index` - The index of the executor.
+    // * `total_executors` - The total number of executor.
+    //
+    // A tuple representing the start and end indices of the tasks assigned to the executor
+    fn calculate_task_range(
+        &self,
+        executor_index: usize,
+        total_executors: usize,
+    ) -> (usize, usize) {
+        let total_tasks = self.entry_csvs.len();
+        let base_tasks_per_executor = total_tasks / total_executors;
+        let extra_tasks = total_tasks % total_executors;
+
+        let start = executor_index * base_tasks_per_executor + min(executor_index, extra_tasks);
+        let end =
+            (executor_index + 1) * base_tasks_per_executor + min(executor_index + 1, extra_tasks);
+
+        (start, min(end, total_tasks))
+    }
+
+    /// Processes a list of CSV files concurrently using executors and aggregates the results.
+    ///
+    /// * `number_executor` - The number of executors to use.
+    ///
+    /// Data flow
+    ///
+    /// 1. Splits the list of CSV files into segments based on the number of available executors.
+    /// 2. A distribution thread loads each CSV file, parses it into `entries`, and sends these to `entries_tx`.
+    /// 3. Each executor receives `entries` from `entries_rx`, requests tasks to Worker, and sends results back through `tree_tx`.
+    /// 4. The processed data from all executors, collected from `tree_rx`, is aggregated into an `AggregationMerkleSumTree`.
+    /// 6. After processing, executors are terminated to release resources.
+    ///
+    async fn create_aggregation_mst(
+        self,
+        number_executor: usize,
+    ) -> Result<AggregationMerkleSumTree<N_ASSETS, N_BYTES>, Box<dyn std::error::Error>>
+    where
+        [usize; N_ASSETS + 1]: Sized,
+        [usize; 2 * (1 + N_ASSETS)]: Sized,
+    {
+        let entries_per_executor = self.entry_csvs.len() / number_executor;
+
+        // Declare channels for internal communication
+        // One of the channels receives parsed data from entry parser to distribute tasks to executor
+        // while the other channel is used the executors that send a result of the tasks.
+        let mut executors = Vec::new();
+        let mut result_collectors = Vec::new();
+
+        let actual_number_of_workers = min(number_executor, self.entry_csvs.len());
+        for i in 0..actual_number_of_workers {
+            let (entries_tx, mut entries_rx) = mpsc::channel(32);
```

Comment:
What's 32? It's probably better to create a named constant for this value.

Reply:
You are right. It should be adjustable for optimizing
I will update for it.

Reply:
Updated at [3e39b10](https://github.com/summa-dev/summa-aggregation/pull/3/commits/3e39b105ff25c1ea2d7e99a187c6c59eff62a6c3)
Now we can adjust with the environment variable `CHANNEL_SIZE`.